clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -692,6 +692,16 @@ def warn_maybe_falloff_nonvoid_function : Warning<
 def warn_falloff_nonvoid_function : Warning<
   "non-void function does not return a value">,
   InGroup<ReturnType>;
+def warn_const_attr_with_pure_attr : Warning<
+  "'const' attribute imposes greater restrictions than 'pure', 'pure' attribute ignored">,
+  InGroup<IgnoredAttributes>;
+def warn_pure_attr_on_cxx_constructor : Warning<
+  "constructor cannot be '%select{pure|const}0', attribute ignored">,
+  InGroup<IgnoredAttributes>;
+def warn_pure_function_returns_void : Warning<
+  "'%select{pure|const}0' attribute on function returning 'void'">,
```

Comment:
I think this should also say `; attribute ignored` and be under the `IgnoredAttribtues` warning group. The attribute has no semantics for a function which returns `void`.